dist/rollbar.js

```diff
@@ -451,8 +451,8 @@ function makeUnhandledStackInfo(
   };
   location.func = errorParser.guessFunctionName(location.url, location.line);
   location.context = errorParser.gatherContext(location.url, location.line);
-  var href = document && document.location && document.location.href;
-  var useragent = window && window.navigator && window.navigator.userAgent;
+  var href = typeof document !== 'undefined' && document && document.location && document.location.href;
```

Comment:
I think document has to be checked first

Reply:
This is a build artifact for a previously merged PR, but aside from that, I'm curious. What would the suggested change look like?

Reply:
var href = document && typeof document !== 'undefined' && document.location && document.location.href;

it makes a difference for python when it's evaluated first, maybe it does not matter for js.

Reply:
It does matter for js. That code will throw a document is not defined exception.

I'm curious again though. In python when document is undefined it will evaluate `true`? If not then`typeof document !== 'undefined'` is superfluous. It will never be reached with undefined.

Reply:
@waltjones there is no `undefined` state in Python, so you will get `NameError` exception regarding the undefined object.

So, in Python, you have to be sure an object (id reference to be precious) is defined before you reference to it.

Reply:
I wouldn't say so.
You can't verify if object/id is (un)defined in Python's frontend. There are some workarounds like try/except or by inspecting internal dictionaries etc. However, there is no official/recommended way to do this.

In my opinion, JS is closer to Ruby or Perl here.

Reply:
I meant roughly the same in that you cannot do `if (foo)` in either language when `foo` is undefined.